File: youShouldCheckOutThisBand/Models/Track.cs
using AutoMapper.Configuration.Annotations;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace youShouldCheckOutThisBand.Models
{
    [NotMapped]
    public class Track:BaseTrack
    {
        public Album Album { get; set; }
        [Ignore]
        public IEnumerable<Artist> Artists {get;set;}     
        public int SumOfVotes
        {
            get
            {
                return UpVotes - DownVotes;
            }
        }

    }
}

File: youShouldCheckOutThisBand/ClientApp/shared/track.ts
//make a track class for type safety

import { Artist } from './artist';

export interface Track {
    
    album: [];
    artists: Artist[];
        sumOfVotes: number;
        href: string;
        previewUrl?: string;
        uri: string;
        name: string;
        upVotes: number;
        downVotes: number;
    
}
